manifests/server/database_ini.pp

```diff
@@ -18,29 +19,35 @@
   $confdir           = $puppetdb::params::confdir,
 ) inherits puppetdb::params {
 
-  # Validate the database connection.  If we can't connect, we want to fail
-  # and skip the rest of the configuration, so that we don't leave puppetdb
-  # in a broken state.
-  #
-  # NOTE:
-  # Because of a limitation in the postgres module this will break with
-  # a duplicate declaration if read and write database host+name are the
-  # same.
-  class { 'puppetdb::server::validate_db':
-    database          => $database,
-    database_host     => $database_host,
-    database_port     => $database_port,
-    database_username => $database_username,
-    database_password => $database_password,
-    database_name     => $database_name,
+  if str2bool($database_validate) {
+    # Validate the database connection.  If we can't connect, we want to fail
+    # and skip the rest of the configuration, so that we don't leave puppetdb
+    # in a broken state.
+    #
+    # NOTE:
+    # Because of a limitation in the postgres module this will break with
+    # a duplicate declaration if read and write database host+name are the
+    # same.
+    class { 'puppetdb::server::validate_db':
+      database          => $database,
+      database_host     => $database_host,
+      database_port     => $database_port,
+      database_username => $database_username,
+      database_password => $database_password,
+      database_name     => $database_name,
+    }
   }
 
+  $ini_setting_require = str2bool($database_validate) ? {
+    false => undef,
+    default  => Class['puppetdb::server::validate_db'],
```

Comment:
Probably not worth doing it this time around, but worth thinking about later. If we ever get around to implementing a defined resource to wrap these ini settings or something, probably better to do it using that resource instead of inifile, since inifile could bleed into other modules and create more edges than necessary in the graph. Extraneous edges == bad sometimes, because it makes for a larger catalog.